### PR TITLE
This issue adds plan-filter into TFaGA

### PR DESCRIPTION
**Pull request by phracek**
Add plan-filter into TFaGA

Closes: #222

### CODE REVIEW

**Comment by phracek** [comment]
[test]

File: README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ See [Testing Farm onboarding guide](https://docs.testing-farm.io/Testing%20Farm/
 | `git_url` | An url to the repository with tmt metadata | url of repo that runs GHA |
 | `git_ref` | A tmt tests branch, tag or commit specifying the desired git revision which will be used for tests | `github.ref` when available, otherwise `master` |
 | `tmt_plan_regex` | A regular expression used to select tmt plans | all |
+| `tmt_plan_filter` | A filter used to select tmt plans | "" |
```
Comment (jamacku) [inline README.md:41]: I would also reference TF documentation of `plan_filter` for more information:

link: https://docs.testing-farm.io/Testing%20Farm/0.1/test-request.html#plan-filter

File: src/action.ts [hunk 2]
```diff
@@ -98,6 +99,14 @@ async function action(pr: PullRequest): Promise<void> {
     ? { name: tmtPlanRegexParsed.data }
     : {};
 
+  // Conditionally include the name attribute only if tmt_plan_Filter is not null
+  const tmtPlanFilterParsed = tmtPlanFilterSchema.safeParse(
+    getInput('tmt_plan_filter')
+  );
+  const tmtPlanFilter = tmtPlanFilterParsed.success
+    ? { name: tmtPlanFilterParsed.data }
```
Comment (jamacku) [inline src/action.ts:107]: Based on TF documentation, `plan_filter` accepts a string, not an object.

https://docs.testing-farm.io/Testing%20Farm/0.1/test-request.html#plan-filter

```suggestion
    ? tmtPlanFilterParsed.data
```

File: src/action.ts [hunk 3]
```diff
@@ -98,6 +99,14 @@ async function action(pr: PullRequest): Promise<void> {
...
+    : {};
```
Comment (jamacku) [inline src/action.ts:108]:
```suggestion
    : '';
```

**Review by jamacku — CHANGES_REQUESTED** [review]
Please include short description in commit message

File: README.md (hunk 1 above)
Comment (phracek) [reply README.md]: Fixed by https://github.com/sclorg/testing-farm-as-github-action/pull/223/commits/9f61ef1cd42b030b982850456ca21d8ad11b6f75

**Comment by spetrosi** [comment]
One question, tmt allows for providing several tags with the filter like `tmt plans ls --filter "tag:mssql,general,postfix"`.
Will that work with `tmt_plan_filter` too? I guess if I set `tmt_plan_filter: "tag:mssql,general,postfix"`.

**Comment by phracek** [comment]
> One question, tmt allows for providing several tags with the filter like `tmt plans ls --filter "tag:mssql,general,postfix"`. Will that work with `tmt_plan_filter` too? I guess if I set `tmt_plan_filter: "tag:mssql,general,postfix"`.

@spetrosi Well as an input we get string, so If you enter string, like you want, we send it to Testing Farm.
As soon as this pull request is merged into main, can you please test it and provide a feedback. Thanks

**Comment by phracek** [comment]
@spetrosi Can you please past link to your plans for an inspiration?

**Comment by spetrosi** [comment]
They are not ready yet, so far I am using a single play https://github.com/linux-system-roles/tft-tests/tree/main/plans.
For testing how tmt filters, I copied this plan and applied different tags to each copy. BTW tmt also supports setting several tags which is good.

**Review by jamacku — APPROVED** [review]
Great work, LGTM

**Comment by spetrosi** [comment]
@phracek I am running TFaGA with `tmt_plan_filter: "tag:general,postfix"`, the workflow returns a warning:

Warning: Unexpected input(s) 'tmt_plan_filter', valid inputs are ['api_key', 'api_url', 'tf_scope', 'git_url', 'git_ref', 'tmt_plan_regex', 'tmt_context', 'tmt_path', 'compose', 'arch', 'variables', 'secrets', 'tmt_hardware', 'copr', 'copr_artifacts', 'github_token', 'create_issue_comment', 'pull_request_status_name', 'update_pull_request_status', 'environment_settings', 'pipeline_settings', 'create_github_summary', 'timeout', 'repo_owner', 'repo_name', 'pr_number', 'commit_sha']

**Pull request merged**